### PR TITLE
Swift 6, SwiftUnicodeDataTables, Other Improvements

### PR DESCRIPTION
**Pull request by STREGA**
- Reworked PDCPlugin so PlaydateKit and CPlaydate are built as part of the product dependencies
- Added simple support for C/C++ modules. Since the PDCPlugin cannot access the Package cSettings/cxxSettings this will not be that useful, but simple C/C++ code modules should work
- Added the libSwiftUnicodeDataTables source files, directly from the SwiftLang repository, as a source target.
- Bumped the package to Swift Language Mode v6 and the Package tools version to swift 6.
- Fixed all Swift 6 deprecations
- Moves helper variables in Package.swift to the bottom of the file. This change will help reduce build errors when users copy the variables from PlaydateKit to their own package.

### CODE REVIEW

**Comment by STREGA** [comment]
Added support for linking SwiftUnicodeDataTables.
In my testing the compiled pdx is the same size as when linked and not linked, when not using any functions that requires the unicode stuff. So this linking is on by default, but a new argument `--disableSwiftUnicodeDataTables` is added just in case there's a problem in the future. The fix is kind of a hack so the Playdate SDK changing could break the hack.

**Comment by STREGA** [comment]
Use the plugins PlaydateSDK path for pdc CLI
Show the build path in output to make finding it easier for users

**Comment by STREGA** [comment]
@finnvoor sorry for so many changes in 1 PR, mostly all related stuff. Confirmed to work on device.

File: Plugins/PDCPlugin/PDCPlugin.swift [hunk 1]
```diff
@@ -214,18 +357,46 @@ struct ModuleBuildRequest {
...
+            if verbose { process.print() }
+            try process.run()
+            process.waitUntilExit()
+            guard process.terminationStatus == 0 else { throw Error.arFailed(exitCode: process.terminationStatus) }
+        }
 
         func pdc(_ arguments: [String]) throws {
             let process = Process()
             process.executableURL = URL(filePath: "\(playdateSDK)/bin/pdc")
-            process.arguments = ["--skip-unknown"] + arguments
```
Comment (finnvoor) [inline Plugins/PDCPlugin/PDCPlugin.swift:228]: any reason why this was removed? Without it the debug symbols are copied into the pix

**Review by finnvoor — APPROVED** [review]
This looks excellent, thanks for working on this. Seems like this also adds support for more than 1 level of dependencies which is great. 

Ideally we'll be able to remove the unicode table target once the Swift team builds them without exceptions. Some people might also want a flag to opt out of linking them to reduce code size, but it seems like code size isn't a huge issue on Playdate.

File: Plugins/PDCPlugin/PDCPlugin.swift (hunk 1 above)
Comment (STREGA) [reply Plugins/PDCPlugin/PDCPlugin.swift:228]: With that flag on, users can't have their own custom file types or any files the pdc doesn't compile. For example 3D models wouldn't be copied.

I'll add an extra step to delete the debug symbols from the finished pdx.

**Comment by STREGA** [comment]
Yup! Specifically I wanted to make sure PlaydateKit could be in a package (game engine) that users can import and use easily. It'll allow people to make and share "RPGKit" and stuff like that.

And yes I agree. This method could possibly cause problems if the toolchain changes anything. Definitely will be ideal to have the official static library available.

**Comment by STREGA** [comment]
Added a step to remove the copied debug symbols. 
Also cleaned up some unnecessary output stuff.

**Pull request merged**